spec/controllers/search_controller_spec.rb

```diff
@@ -0,0 +1,25 @@
+RSpec.describe SearchController, type: :controller do
+  describe 'redirect to 404 if page number more than total pages' do
+    let(:search_form_params) do
+      {
+        'advice_method' => SearchForm::ADVICE_METHOD_FACE_TO_FACE,
+        'postcode' => 'EC1N 2TD'
+      }
+    end
+    let(:firm_repository) { double }
+    let(:search_result) { double }
+
+    before do
+      allow(FirmRepository).to receive(:new).and_return(firm_repository)
+      allow(firm_repository).to receive(:search).and_return(search_result)
+      allow(search_result).to receive(:total_pages).and_return(0)
+    end
+
+    it 'redirects to 404' do
+      VCR.use_cassette(:geocode_search_form_postcode) do
+        get :index, locale: :en, page: 1, search_form: search_form_params
+        expect(response).to redirect_to '/404.html'
+      end
+    end
+  end
+end
```

Comment:
I like that you've added tests for a controller that didn't **but obviously should have had tests**. Please would you add tests for the other paths of the index method.